test/contracts/workers/set_worker.js

```diff
@@ -0,0 +1,110 @@
+  const opsAddress          = accounts[1],
+        worker1Address       =  accounts[2],
+        worker2Address       =  accounts[3],
+        height1  = new workers_utils.bigNumber(500),
+        height2  = new workers_utils.bigNumber(1000);
+        
+  before(async () => {
+
+    workers = await Workers.new();
+    assert.ok(await workers.setOpsAddress(opsAddress));    
+
+  });
+
+
+  it('fails to set worker if worker address is 0', async () => {
+
+    await workers_utils.utils.expectThrow(workers.setWorker.call(0, height1, { from: opsAddress }));    
+
+  });
+
+
+  it('fails to set worker if _deactivationHeight is equal to current block number', async () => {    
+
+    await workers_utils.utils.expectThrow(workers.setWorker.call(0, web3.eth.blockNumber, { from: opsAddress }));    
+
+  });
+
+
+  it('fails to set worker if _deactivationHeight is less than current block number', async () => {    
+
+    await workers_utils.utils.expectThrow(workers.setWorker.call(0, 0, { from: opsAddress }));    
+
+  });
+
+
+  it('fails to set worker if sender is not opsAddress', async () => {
+
+    deactivationHeight = web3.eth.blockNumber + height1.toNumber();    
+    await workers_utils.utils.expectThrow(workers.setWorker.call(worker1Address, deactivationHeight, { from: accounts[5] }));    
+
+  });
+
+
+  it('pass to set worker if worker is not already set', async () => {
+    
+    deactivationHeight = web3.eth.blockNumber + height1.toNumber();
+    assert.ok(await workers.setWorker.call(worker1Address, deactivationHeight, { from: opsAddress }));
+    response = await workers.setWorker(worker1Address, deactivationHeight, { from: opsAddress });
+    assert.equal(await workers.isWorker.call(worker1Address), true);
+    workers_utils.checkWorkerSetEvent(response.logs[0], deactivationHeight, height1.toNumber()-1);
+    
+    deactivationHeight = web3.eth.blockNumber + height2.toNumber();
+    assert.ok(await workers.setWorker.call(worker2Address, deactivationHeight, { from: opsAddress }));
+    response = await workers.setWorker(worker2Address, deactivationHeight, { from: opsAddress });
+    assert.equal(await workers.isWorker.call(worker2Address), true);
+    workers_utils.checkWorkerSetEvent(response.logs[0], deactivationHeight, height2.toNumber()-1);
+    workers_utils.utils.logResponse(response, 'Worker.setWorker');
+
+  });
+
+
+  it('pass to set worker if worker was already set', async () => {
+    
+    deactivationHeight = web3.eth.blockNumber + height2.toNumber();
+    assert.ok(await workers.setWorker.call(worker1Address, deactivationHeight, { from: opsAddress }));
+    response = await workers.setWorker(worker1Address, deactivationHeight, { from: opsAddress });
+    assert.equal(await workers.isWorker.call(worker1Address), true);
+    workers_utils.checkWorkerSetEvent(response.logs[0], deactivationHeight, height2.toNumber()-1);
+    workers_utils.utils.logResponse(response, 'Worker.setWorker(update)');
+
+  });
+
```

Comment:
possible extra test could be to set the deactivation first high; and then set it lower, but still active;  perhaps worth putting in an issue as a small task for later.

Reply:
Added